enterprise_access/apps/api/v1/views/utils.py

```diff
@@ -1,19 +1,59 @@
 """
 Utilities for REST API View and Viewset classes.
 """
-from rest_framework.pagination import PageNumberPagination
+from django.conf import settings
+from edx_rest_framework_extensions.paginators import DefaultPagination
 
 
-class PaginationWithPageCount(PageNumberPagination):
+class PaginationWithPageCount(DefaultPagination):
     """
-    A PageNumber paginator that adds the total number of pages to the paginated response.
+    A PageNumber paginator that adds the total number of pages
+    and the current page to the paginated response.
     """
 
     page_size_query_param = 'page_size'
+    # The configured `PAGE_SIZE` in Django settings must be used as the default page size as opposed
+    # to relying on the default `page_size` specified via `DefaultPagination` in order to maintain
+    # backwards compatibility with existing API clients.
+    page_size = settings.REST_FRAMEWORK.get('PAGE_SIZE')
     max_page_size = 500
 
-    def get_paginated_response(self, data):
-        """ Adds a ``num_pages`` field into the paginated response. """
-        response = super().get_paginated_response(data)
-        response.data['num_pages'] = self.page.paginator.num_pages
-        return response
+
+class OptionalPaginationWithPageCount(PaginationWithPageCount):
+    """
+    Optionally allows callers to disable pagination by setting the (configurable) `no_page` query param.
+    """
+    no_page_query_param = 'no_page'
+
+    def formatted_no_page_query_param(self, request):
+        """
+        Parses the query parameters for `self.no_page_query_param` and transforms
+        its value to a lowercase string, if present.
+        """
+        no_page_query_param = request.query_params.get(self.no_page_query_param)
+        formatted_no_page_query_param = no_page_query_param.lower() if no_page_query_param else no_page_query_param
+        return formatted_no_page_query_param
```

Comment:
code-golf nit: you could default the `get()` to return an empty string, or use `or ''` to do something like e.g.
```python
param_value = request.query_params.get(self.no_page_query_param) or ''
return param_value.lower()
```
or if we believe `no_page` will never be set explicitly as null in the query params, simply
```python
return request.query_params.get(self.no_page_query_param, '').lower()
```